### ISSUE TITLE
Antialiasing and MultipleRenderTargets

### ISSUE THREAD

**Issue by RenaudRohlinger** [vanruesc/postprocessing#333]
Hello! I was wondering if it was possible to use multiple render targets as a RenderPass and then use an antialiasing technic such as MSAA EffectPass to try to antialias the result? MSAA should not be possible on a texture as far as I know, but I am a beginner in AA paradigms. That is why I am writing this issue as it seems there is a lot of interesting AA, maybe you might have already tried?

The only solution I am seeing otherwise would be to blit and multisample each layout of the MRT.

If an MRT is provided in a ShaderPass, the next pass could technically antialias it? And is the RenderPass required in that case?

**Comment by vanruesc** [comment]
Hi,

I haven't really touched MRT yet, but it looks like it's currently not possible to use `WebGLMultipleRenderTargets` with `WebGLMultisampleRenderTarget` due to the way it's implemented in `three`. This is something that should be addressed since it's [supported in WebGL 2](https://www.khronos.org/webgl/public-mailing-list/public_webgl/1609/msg00084.php).

> If an MRT is provided in a ShaderPass, the next pass could technically antialias it? And is the RenderPass required in that case?

Individual textures of a `WebGLMultipleRenderTargets` instance could be antialiased with a post effect like SMAA, but MSAA would be better overall if it was available. A `RenderPass` is always required if you want to render meshes, even if it's just a plane with a texture. I'm currently working on the API to support explicit buffer management which would allow you to use a plain texture as input for a pass, if that's what you want to do.

> MSAA should not be possible on a texture as far as I know

Right, MSAA works based on geometry and depth; it's not a post effect and it doesn't affect texture details.

**Comment by RenaudRohlinger** [comment]
Thanks for the explanation @vanruesc! I did try several times to implement MSAA + MRT directly in threejs without success as WebGLTextures has become pretty hard to understand especially with the latest change of the MSAA for VR.

I think explicit buffer management would be perfect for the moment. I will use SMAA as your implementation looks pretty solid.
I just made a minimalist reproduction of threejs MRT + the SMAA of postprocessing. I think it has some encoding issues as I wanted to give RGBFormat but TextureEffect is expecting, either way, sRGB or Linear, I am not sure how I should send the result of my render target to the SMAA Pass. If in the meantime you have any idea how to solve this encoding issue that would be awesome!

Example MRT + Postprocessing SMAA:
https://codesandbox.io/s/wonderful-spence-0tuwv?file=/src/index.js
